### PR TITLE
Create tests for PreferenceModel

### LINKED ISSUE #113: Create tests for PreferenceModel

### PR DESCRIPTION
**Pull request by NoB0**
Fixes #113

Note: the test `test_get_slot_preference` depends on PR #117.

### CODE REVIEW

File: tests/simulator/user_modeling/test_preference_model.py [hunk 1]
```diff
@@ -40,68 +37,52 @@ def preference_model_sip() -> PreferenceModel:
...
-    preference = preference_model_sip.get_slotvalue_preference(
-        "GENRE", "Comedy"
-    )
-    assert -1 <= preference <= 1
-
-
-def test_get_slotvalue_preference_pkg(preference_model_pkg) -> None:
-    preference = preference_model_pkg.get_slotvalue_preference("GENRE", "Drama")
-    assert preference == 0.6
+def test_get_item_preference_error(preference_model: PreferenceModel) -> None:
```
Comment (kbalog) [inline tests/simulator/user_modeling/test_preference_model.py:56]: Should we indicate in the method name that we're testing preference for a non-existing item? e.g., `test_get_item_preference_nonexisting_item`

File: tests/simulator/user_modeling/test_preference_model.py [hunk 2]
```diff
@@ -40,68 +37,52 @@ def preference_model_sip() -> PreferenceModel:
...
+        "GENRE", "Animation"
+    )
+    assert preference == 1.0 or preference == -1.0
 
 
-def test_get_item_preference_pkg(preference_model_pkg):
-    # When (Both Drama and Thriller are favored as 1)
-    preference = preference_model_pkg.get_item_preference("100")
-    assert preference == 1.0
+def test_get_slot_value_preference_error(
```
Comment (kbalog) [inline tests/simulator/user_modeling/test_preference_model.py:68]: Similarly, here, perhaps name it `test_get_slot_value_preference_nonexisting_slot`?

File: tests/simulator/user_modeling/test_preference_model.py [hunk 3]
```diff
@@ -40,68 +37,52 @@ def preference_model_sip() -> PreferenceModel:
...
+            "GENRE"
+        )
+    )
+    assert preference == 1
 
 
-# TODO Write tests for get_item_preference(), get_slotvalue_preference,
-# and get_slot_preference()
-# See: https://github.com/iai-group/UserSimCRS/issues/21
+def test_get_slot_preference_error(preference_model: PreferenceModel) -> None:
```
Comment (kbalog) [inline tests/simulator/user_modeling/test_preference_model.py:86]: Name it `test_get_slot_preference_nonexisting_slot`?

**Review by kbalog — APPROVED** [review]
LGTM (w/ some suggestions for naming the test methods for consideration)

**Comment by NoB0** [comment]
I changed the name of the tests where an exception is expected for better readability.

**Pull request merged**